pkg/webhook/webhook.go

```diff
@@ -199,7 +195,90 @@ func (wh *webhook) validate(ar *v1beta1.AdmissionReview) *v1beta1.AdmissionRespo
 			Status: metav1.StatusFailure, Code: http.StatusForbidden, Reason: "PVC with cloned volumes can't be deleted",
 			Message: fmt.Sprintf("pvc %q has '%v' cloned volume(s)", pvc.Name, len(cStorVolumes.Items)),
 		}
+		return response
+	}
+	return response
+}
+
+// validatePVCCreaterequest validates persistentvolumeclaim(PVC) create request
+func (wh *webhook) validatePVCCreaterequest(req *v1beta1.AdmissionRequest) *v1beta1.AdmissionResponse {
+	response := &v1beta1.AdmissionResponse{}
+	response.Allowed = true
+
+	var pvc corev1.PersistentVolumeClaim
+	err := json.Unmarshal(req.Object.Raw, &pvc)
+	if err != nil {
+		glog.Errorf("Could not unmarshal raw object: %v, %v", err, req.Object.Raw)
+		response.Allowed = false
+		response.Result = &metav1.Status{
+			Status:  metav1.StatusFailure,
+			Code:    http.StatusBadRequest,
+			Reason:  metav1.StatusReasonBadRequest,
+			Message: err.Error(),
+		}
+		return response
+	}
+
+	// If snapshot.alpha.kubernetes.io/snapshot annotation represents the clone pvc
+	// create request
+	snapname := pvc.Annotations["snapshot.alpha.kubernetes.io/snapshot"]
+	if len(snapname) == 0 {
+		return response
+	}
+
+	glog.Infof("AdmissionReview for creating a clone volume Kind=%v, Namespace=%v Name=%v UID=%v patchOperation=%v UserInfo=%v",
```

Comment:
does Infof logs comes in regular cases? If so, we need to change the level so that this won't come regularly

Reply:
It will log only if it is a clone volume creation

Reply:
we don't need this even for every clone creation request, unless there is failure